lib/rack/files.rb

```diff
@@ -70,75 +71,111 @@ def serving(request, path)
       headers[CONTENT_TYPE] = mime_type if mime_type
 
       # Set custom headers
-      @headers.each { |field, content| headers[field] = content } if @headers
-
-      response = [ 200, headers ]
+      headers.merge!(@headers) if @headers
 
+      status = 200
       size = filesize path
 
-      range = nil
       ranges = Rack::Utils.get_byte_ranges(request.get_header('HTTP_RANGE'), size)
-      if ranges.nil? || ranges.length > 1
-        # No ranges, or multiple ranges (which we don't support):
-        # TODO: Support multiple byte-ranges
-        response[0] = 200
-        range = 0..size - 1
+      if ranges.nil?
+        # No ranges:
+        ranges = [0..size - 1]
       elsif ranges.empty?
         # Unsatisfiable. Return error, and file size:
         response = fail(416, "Byte range unsatisfiable")
         response[1]["Content-Range"] = "bytes */#{size}"
         return response
-      else
-        # Partial content:
+      elsif ranges.size == 1
+        # Partial content, single range:
+        status = 206
         range = ranges[0]
-        response[0] = 206
-        response[1]["Content-Range"] = "bytes #{range.begin}-#{range.end}/#{size}"
+        headers["Content-Range"] = "bytes #{range.begin}-#{range.end}/#{size}"
         size = range.end - range.begin + 1
+      else
+        # Partial content, multiple ranges:
+        status = 206
+        headers[CONTENT_TYPE] = "multipart/byteranges; boundary=#{MULTIPART_BOUNDARY}"
```

Comment:
Sorry, I only have a short time to review this, but I'm excited to see this PR.

Is `AaB03x` sufficient as a boundary?

Reply:
Yes, why not? The same boundary is used for multipart forms https://github.com/rack/rack/blob/93dfcdf46084760079514c858ae7391e90a7821e/lib/rack/multipart.rb#L14